bigframes/operations/semantics.py

```diff
@@ -296,6 +300,9 @@ def cluster_by(
                 "It must be greater than 1."
             )
 
+        if not self._confirm_operation(len(self._df)):
+            return None
```

Comment:
I'm a bit confused by the change of return type. This would make type checking much more difficult. Let's raise an exception saying the operation was cancelled instead.

Reply:
SG!